### PR TITLE
(CDAP-4864) Make MetadataRecord backwards-compatible. Deprecate the c…

### PR DESCRIPTION
**Pull request by bdmogal**
…onstructor that defaults to MetadataScope#USER instead of removing it.

### CODE REVIEW

File: cdap-proto/src/main/java/co/cask/cdap/proto/metadata/MetadataRecord.java [hunk 1]
```diff
@@ -48,6 +48,16 @@ public MetadataRecord(MetadataRecord other) {
     this(other.getEntityId(), other.getScope(), other.getProperties(), other.getTags());
   }
 
+  /**
+   * Returns a new {@link MetadataRecord} in {@link MetadataScope#USER}.
+   * Deprecated as of 3.3. Please use {@link #MetadataRecord(Id.NamespacedId, MetadataScope, Map, Set)} instead.
```
Comment (anwar6953) [inline cdap-proto/src/main/java/co/cask/cdap/proto/metadata/MetadataRecord.java]: I think specify 3.3.1, since it wasn't deprecated in 3.3.0.

**Comment by bdmogal** [comment]
@anwar6953 addressed comment. Please take another look.

**Comment by anwar6953** [comment]
lgtm

**Comment by bdmogal** [comment]
The code that is added here is never called across CDAP. Hence merged without a build. Ensured that checkstyle passes.

**Pull request merged**